lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -118,11 +118,20 @@ def fetch_pod_metadata(namespace_name, pod_name)
       @stats.bump(:pod_cache_api_updates)
       log.trace("parsed metadata for #{namespace_name}/#{pod_name}: #{metadata}")
       @cache[metadata['pod_id']] = metadata
-    rescue StandardError => e
-      @stats.bump(:pod_cache_api_nil_error)
-      log.debug "Exception '#{e}' encountered fetching pod metadata from Kubernetes API #{@apiVersion} endpoint #{@kubernetes_url}"
-      {}
-    end
+      rescue KubeException => e
+        if e.error_code == 401
+          # recreate client to refresh token
+          log.info("Encountered '401 Unauthorized' exception, recreating client to refresh token")
+          create_client()
+        else
+          log.error "Exception '#{e}' encountered fetching pod metadata from Kubernetes API #{@apiVersion} endpoint #{@kubernetes_url}"
+          @stats.bump(:pod_cache_api_nil_error)
+        end
```

Comment:
This looks like it needs to return `{}` after this end as callers are expecting a map.  I'm not certain otherwise what is the result upon error

Reply:
Ah I see... I was wondering what the `{}` in the other rescue was for... will fix..